routes/cart-router.js

```diff
@@ -0,0 +1,80 @@
+const express = require("express");
+const router = express.Router();
+
+const { CartItem } = require('../models/cart.model');
+const { WishlistItem } = require('../models/wishlist.model');
+const { extend } = require('lodash');
+
+router.get("/summary", async (req, res) => {
+  const cart = await CartItem.find({});
+  res.status(200).json({ cartLength: cart.length })
+})
+
+router.route("/")
+  .get(async (req, res) => {
+    try {
+      const cart = await CartItem.find({}).populate("product");
+      res.status(200).json({ cart, success: true })
+    }
+    catch (err) {
+      res.status(500).json({ success: false, errorMessage: err.message })
+    }
+  })
+  .post(async (req, res) => {
+    try {
+      const product = req.body;
+      const isPresent = await CartItem.exists({ product: product.product._id });
```

Comment:
I don't think you need to call the API for checking the product whether its exists or not in the cart. Check it on frontend.Unnecessary API call right?

Reply:
No, I am not calling the wishlist and cart API when the app Is loaded. So when the app is loaded I am checking in the backend itself if it is in the wishlist or cart. So that's why did this way.